sphinx/ext/autodoc.py

```diff
@@ -171,13 +171,15 @@ def __init__(self, names):
             #     set(['a', 'd'])
             self.base_packages.add(n.split('.')[0])
         self.mocked_modules = []  # type: List[str]
-        self.orig_meta_path = sys.meta_path
         # enable hook by adding itself to meta_path
         sys.meta_path = sys.meta_path + [self]
 
     def disable(self):
         # restore original meta_path to disable import hook
-        sys.meta_path = self.orig_meta_path
+        for idx, importer in enumerate(sys.meta_path):
+            if importer is self:
+                del sys.meta_path[idx]
+                break
```

Comment:
I meant "only self must be removed and not the self and all finders/importers after it".
I tried to be very cautious here. Your suggestion may work but I was not able to find any requirements for comparison behavior of finders in PEPs. So by using `remove` we will rely on a [rich] comparison implementation of objects in  `sys.meta_path` or if it is not defined `id` is implicitly used. Therefore it looks to me that it is safer to compare explicitly by `id`.
Another concern was that `sys.meta_path` may not be Python list but some other sequence so I did not use list comprehension. But I just checked  PEP 302 for that and `sys.meta_path` must be a Python list so I suggest to change the code above for the following list comprehension:
``` sys.meta_path = [i for i in sys.meta_path if i is not self]```

How about that?